Add templates for promoting features (#3)
istio/enhancements: .github/ISSUE_TEMPLATE/stable_feature.md

---
name: Stable Feature Requirements
about: Requirements for promoting a feature to stable
---

> This page lists the requirements for promoting a feature to stable. Please check off and document the steps as they are completed.

Reviewer: On each of these we should note that they include all requirements for the previous step. So maybe in stable there is a place to put a link to the issue for beta, etc. So we can chain all they way back and see all the items have been checked off?

One thing to think about is how to manage growing the checklist while features are in the process. If we add a new item to beta checklist, but a feature is already beta, how do we make sure they did it before going to stable?

**Feature Name:** 
--- 

### Requirements: 

**Tests**

- [ ] Test coverage is >= 90%
- [ ] Automated tests are in place to prevent performance regressions. 

**Performance**

> - [ ] Latency, throughput, and scale are quantified and documented. 

Reviewer: ```suggestion
- [ ] Latency, throughput, and scalability are quantified and documented. 
```

Reviewer: Documented on istio.io?

**Bugs**

> - [ ] Feature has no outstanding P0 or P1 bugs. 

Reviewer 2: No P1 bugs? Good luck...

> **Performance**

Reviewer: ```suggestion
**Promotion**
```

Author: Fixed. Thanks.

Reviewer: Doesn't look fixed.

I think we should actually call this section Approvals or Reviews though.

> - [ ] The decision to promote a feature is presented in the roadmap of the appropriate working group. 

Reviewer: ```suggestion
- [ ] The decision to promote a feature is presented in the roadmap of the appropriate working groups.
```

BTW we should be consistent and use either workgroups or working groups.

Author: Fixed. Thanks.

> - [ ] TOC has approved promotion. 

Reviewer: ```suggestion
- [ ] TOC has approved promotion to stable.
```

Reviewer: These lines are not consistent across the checklists, should make them consistent (TOC has approved promotion to alpha/beta/stable)

Author: Fixed. Thanks.

---
name: Stable Feature Requirements
about: Requirements for promoting a feature to stable
---

This page lists the requirements for promoting a feature to stable. Promotion to stable must meet all requirements of promotion to beta. Please check off and document the steps as they are completed.

**Feature Name:** 

> **Promotion to Alpha:**

Reviewer: Alpha Checklist

Author: Done

> **Promotion to Beta:**

Reviewer: Beta Checklist

Author: Done

**Relevant Documentation:**

```
```

--- 

### Requirements: 

**Tests**

> - [ ] Automated tests are in place to prevent regressions. 

Reviewer: I'm not sure how this one is actionable, how would a developer verify this? What is new vs. beta here?

---
name: Stable Feature Requirements
about: Requirements for promoting a feature to stable
---

> This page lists the requirements for promoting a feature to stable. Promotion to stable includes all requirements of promotion to beta. Please check off and document the steps as they are completed.

Reviewer 3: ```suggestion
This page lists the requirements for promoting a feature to stable. Promotion to stable must meet all requirements of promotion to beta. Please check off and document the steps as they are completed.
```

**Feature Name:** 

**Promotion to Alpha:**

**Promotion to Beta:**

**Relevant Documentation:**

```
```

--- 

### Requirements: 

**Tests**

- [ ] Automated tests are in place to prevent regressions. 

**Performance**

- [ ] Latency, throughput, and scalability are quantified and documented on
	istio.io. 

**Bugs**

> - [ ] Feature has no known major issues. 

Reviewer 3: What's a major issue?

Author: Whatever the owning work group decides it is. Say crashes, feature not working, feature not working in certain scenarios... this used to be called P0/P1, but there's no such thing as a P1/P0 bug in terms of a feature (those are release level issues).
